dockstore-webservice/src/main/java/io/dockstore/webservice/resources/NotificationResource.java

```diff
@@ -0,0 +1,110 @@
+    private final NotificationDAO notificationDAO;
+    private final SessionFactory sessionFactory;
+
+    // constructor
+    public NotificationResource(SessionFactory sessionFactory) {
+        this.notificationDAO = new NotificationDAO(sessionFactory);
+        this.sessionFactory = sessionFactory;
+    }
+
+    // get a notification by its id
+    @GET
+    @Path("/notifications/{id}")
+    @UnitOfWork
+    @ApiOperation(value = "Return the notification with given id.", notes = "NO Authentication", responseContainer = "List", response = Notification.class)
+    public Notification getNotification(@PathParam("id") Long id) {
+        Notification notification = notificationDAO.findById(id);
+        throwErrorIfNull(notification);
+        return notification;
+    }
+
+    // get all active notifications
+    @GET
+    @Path("/notifications")
+    @UnitOfWork
+    @ApiOperation(value = "Return all active notifications", notes = "NO Authentication", responseContainer = "List", response = Notification.class)
+    public List<Notification> getActiveNotifications() {
+        return notificationDAO.getActiveNotifications();
+    }
+
+    // post a new notification
+    @POST
+    @Path("/notifications")
+    @UnitOfWork
+    @RolesAllowed({"curator", "admin"})
+    @ApiOperation(value = "Create a notification", authorizations = {@Authorization(value = JWT_SECURITY_DEFINITION_NAME)},
+            notes = "Curator/admin only", response = Notification.class)
+    public Notification createNotification(@ApiParam(value = "Notification to create", required = true) Notification notification) {
+        long id = notificationDAO.create(notification);
+        return notificationDAO.findById(id);
+    }
+
+    // delete a notification by its id
+    @DELETE
+    @Path("/notifications/{id}")
+    @UnitOfWork
+    @RolesAllowed({ "curator", "admin" })
+    @ApiOperation(value = "Delete a notification", authorizations = {@Authorization(value = JWT_SECURITY_DEFINITION_NAME)}, notes = "Curator/admin only")
+    public void deleteNotification(@ApiParam(value = "Notification to delete", required = true) @PathParam("id") Long id) {
+        Notification notification = notificationDAO.findById(id);
+        throwErrorIfNull(notification);
+        notificationDAO.delete(notification);
+    }
+
+    // update an existing notification by its id
+    @PUT
+    @Path("/notifications/{id}")
+    @UnitOfWork
+    @RolesAllowed({ "curator", "admin" })
+    @ApiOperation(value = "Update a notification", authorizations = {@Authorization(value = JWT_SECURITY_DEFINITION_NAME)},
+            notes = "Curator/admin only", response = Notification.class)
+    public Notification updateNotification(@ApiParam(value = "Notification to update", required = true) @PathParam("id") long id,
+                                           @ApiParam(value = "Updated version of notification", required = true) Notification notification) {
+        return notificationDAO.update(notification);
```

Comment:
You need to validate the notification with `{id}` exists, calling `throwErrorIfNull`

Also, this gets sort of weird and I've never found an entirely satisfactory pattern for this, but you need to make sure notification.id matches `{id}`. Otherwise a user could do `PUT /notifications/1`, but the if the notification object had id 2, it would update the one with 2.